subprojects/client-protocol/src/main/java/org/gradle/profiler/client/protocol/serialization/MessageProtocolHandler.java

```diff
@@ -0,0 +1,55 @@
+package org.gradle.profiler.client.protocol.serialization;
+
+import org.gradle.profiler.client.protocol.Connection;
+import org.gradle.profiler.client.protocol.messages.*;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.*;
+
+public class MessageProtocolHandler {
+    private final String peerName;
+    private final Connection connection;
+    private final ExecutorService executorService;
+
+    public MessageProtocolHandler(String peerName, Connection connection) {
+        this.peerName = peerName;
+        this.connection = connection;
+        this.executorService = Executors.newCachedThreadPool();
```

Comment:
I suppose this class should be closable as well and shutdown the `executorService` on `close`.

Reply:
Good point, but I would like to avoid Closeables, since it's easy to forget to close it. I will use `CompletableFuture.supplyAsync` that uses common thread pool. Should do the trick in that case.

Reply:
Is this a good idea, I wonder? I never used the common thread pool, but my expectation would be is that we have no control over it, and if someone else fills it up with requests, our measurements might suffer. For this reason a private executor service sounds better for me.